### PR TITLE
Converting Integration tests to Local Server tests

### PR DESCRIPTION
**Pull request by domesticmouse**
Initial sketch of how I am thinking about crafting non integration tests out of the current integration test suite.

PTAL @markmcd

### CODE REVIEW

File: src/test/java/com/google/maps/DirectionsApiTest.java [hunk 1]
```diff
@@ -26,89 +26,108 @@
 import com.google.maps.model.TransitRoutingPreference;
 import com.google.maps.model.TravelMode;
 import com.google.maps.model.Unit;
+import java.util.ArrayList;
```
Comment (markmcd) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: nit: this looks like it's in the wrong place

File: src/test/java/com/google/maps/LocalTestServerContext.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.google.maps;
+
+import static org.junit.Assert.assertEquals;
```
Comment (markmcd) [inline src/test/java/com/google/maps/LocalTestServerContext.java]: nit: import ordering should follow style guide at go/javastyle (you should be able to set this in your IDE too)

File: src/test/java/com/google/maps/DirectionsApiTest.java [hunk 3]
```diff
@@ -26,89 +26,108 @@
...
   @Test
   public void testGetDirections() throws Exception {
-    DirectionsResult result = DirectionsApi.getDirections(context, "Sydney, AU",
-        "Melbourne, AU").await();
-    assertNotNull(result.routes);
-    assertNotNull(result.routes[0]);
-    assertThat(result.routes[0].overviewPolyline.decodePath().size(), not(0));
-    assertTrue(result.routes[0].legs[0].startAddress.startsWith("Sydney NSW"));
-    assertTrue(result.routes[0].legs[0].endAddress.startsWith("Melbourne VIC"));
+    try (LocalTestServerContext sc = new LocalTestServerContext(getDirectionsResponse)) {
```
Comment (markmcd) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: nice idea 👍

File: src/test/java/com/google/maps/DirectionsApiTest.java [hunk 4]
```diff
@@ -26,89 +26,108 @@
...
-        .departureTime(now)
-        .await();
-
-    assertNotNull(result.routes);
-    assertNotNull(result.routes[0]);
-    assertNotNull(result.routes[0].legs);
-    assertNotNull(result.routes[0].legs[0]);
-    assertNotNull(result.routes[0].legs[0].arrivalTime);
-    assertNotNull(result.routes[0].legs[0].departureTime);
+    try(LocalTestServerContext sc = new LocalTestServerContext(responseTimesArePopulatedCorrectly)) {
```
Comment (markmcd) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: nit: space after try (there are a bunch of these)

File: src/test/java/com/google/maps/DirectionsApiTest.java [hunk 5]
```diff
@@ -132,14 +155,20 @@ public void testTorontoToMontreal() throws Exception {
...
-    DirectionsResult result = DirectionsApi.newRequest(context)
-        .origin("Toronto")
-        .destination("Montreal")
-        .avoid(RouteRestriction.HIGHWAYS)
-        .mode(TravelMode.BICYCLING)
-        .await();
-
-    assertNotNull(result.routes);
+    try(LocalTestServerContext sc = new LocalTestServerContext(
+        "{\"routes\": [{}],\"status\": \"OK\"}")){
```
Comment (markmcd) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: one day java will support multiline here-strings 😩

File: src/test/java/com/google/maps/GeocodingApiTest.java [hunk 6]
```diff
@@ -36,60 +27,86 @@
...
+          .place(placeID)
+          .await();
+      checkSydneyResult(results);
+
+      sc.assertParamValue(placeID, "place_id");
+    }
   }
 
   @Test
   public void testAsync() throws Exception {
```
Comment (markmcd) [inline src/test/java/com/google/maps/GeocodingApiTest.java]: This is a weird test to have in GeocodingApiTests - maybe it belongs it a context or pending result test? Not particularly relevant to your change though so feel free to ignore.

File: src/test/java/com/google/maps/DirectionsApiTest.java (hunk 1 above)
Comment (domesticmouse) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: I'd agree, but [google-java-format](https://github.com/google/google-java-format) seems happy with it.

File: src/test/java/com/google/maps/LocalTestServerContext.java (hunk 2 above)
Comment (domesticmouse) [inline src/test/java/com/google/maps/LocalTestServerContext.java]: Turns out there is tooling for it now...

File: src/test/java/com/google/maps/GeocodingApiTest.java (hunk 6 above)
Comment (domesticmouse) [inline src/test/java/com/google/maps/GeocodingApiTest.java]: I'll have some de-dupe and re-org work to do on this test suite next Q

**Pull request merged**